### PR TITLE
rustfmt all the things

### PR DESCRIPTION
**Pull request by saghm**
I haven't had a chance to completely look this over, but I'll add comments if I see anything noteworthy

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ license = "Apache-2.0"
 name = "mongodb"
 readme = "README.md"
 repository = "https://github.com/mongodb-labs/mongo-rust-driver-prototype"
-version = "0.3.3"
+version = "0.3.4"
```
Comment (saghm) [inline Cargo.toml]: Accidentally didn't push this to master before opening the PR, but it's pushed now

File: src/auth.rs [hunk 2]
```diff
@@ -61,7 +61,8 @@ impl Authenticator {
         let bytes = format!("n,,{}", message).into_bytes();
         let binary = Binary(Generic, bytes);
 
-        let start_doc = doc! {
+        let start_doc =
+            doc! {
             "saslStart" => 1,
```
Comment (saghm) [inline src/auth.rs]: Looks like rustfmt wants to move the macro invocation to a new line but then doesn't indent everything else. I'll fix that here and elsewhere and push a new commit.

File: src/auth.rs [hunk 3]
```diff
@@ -154,8 +171,10 @@ impl Authenticator {
 
         // Sanity check
         if client_key.len() != client_signature.len() {
-            return Err(DefaultError(String::from("Generated client key and/or client signature \
-                                                  is invalid")));
+            return Err(DefaultError(String::from(
+                "Generated client key and/or client signature \
```
Comment (saghm) [inline src/auth.rs]: This can be collapsed back to one line now.

File: src/coll/batch.rs [hunk 4]
```diff
@@ -153,16 +175,28 @@ impl Batch {
             }
             Batch::Update(ref mut models) => {
                 match model {
-                    WriteModel::ReplaceOne { filter, replacement: update, upsert } |
-                    WriteModel::UpdateOne { filter, update, upsert } => {
+                    WriteModel::ReplaceOne {
```
Comment (saghm) [inline src/coll/batch.rs]: Okay, this is weird and unnecessary. I guess rustfmt isn't as improved as I hoped.

File: src/coll/error.rs [hunk 5]
```diff
@@ -114,15 +114,22 @@ impl fmt::Display for BulkWriteException {
...
 
         match self.request {
             Some(ref request) => try!(write!(fmt, "Failed to execute request {:?}\n.", request)),
-            None => try!(write!(fmt, "No additional error information was received.\n")),
+            None => {
+                try!(write!(
+                    fmt,
+                    "No additional error information was received.\n"
+                ))
+            }
```
Comment (kyeah) [inline src/coll/error.rs]: wait, why does it reformat this line, but not the longer one above it? 🤔

**Comment by saghm** [comment]
I'll merge this once #238 is merged to save us a crazy rebase.

**Pull request merged**